canonical-web-and-design/maas-ui
# feat(ui): add Authentication step to Add LXD VM host flow

## Done

- Split `AddLxd` component into three separate forms as per the [design](https://app.zeplin.io/project/610abd78c381eeafb70f5b43/screen/6127bcc601576ca34a8d2ae7)
  1. `CredentialsForm` for the base LXD details
  2. `AuthenticationForm` for choosing to either trust a generated certificate or use a password
  3. `SelectProjectForm` for selecting a project from an authenticated LXD server

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the KVM list and click "Add KVM"
- Choose LXD as the type, input an invalid power address and click "Next". Ignore providing credentials, that will be handled in a separate PR.
- Check that you are taken to a new form where you can choose to either trust a certificate or provide a password. The trust certificate part will be handled in a separate PR.
- Choose to provide a password, input anything and click "Next"
- You should get booted back to the start with an error about being unable to reach the LXD server
- This time input a real power address and click "Next" then input an incorrect password and click "Next" again
- You should be taken to the project selection screen.
- Select any project and click "Next" again
- You should once again be booted to the start, with an error about an incorrect password
- Run through the steps again, this time using a correct password (or alternatively no password if none has been set on the LXD server) and check that you can successfully create a LXD KVM.

## Fixes

Fixes canonical-web-and-design/app-squad#252

## Screenshot
![Peek 2021-09-10 12-52](https://user-images.githubusercontent.com/25733845/132791916-0b991b29-93d6-4fac-82d8-c12b658d3bbe.gif)
